src/main/java/it/gov/pagopa/reward/notification/service/csv/out/retrieve/Initiative2ExportRetrieverServiceImpl.java

```diff
@@ -56,34 +58,33 @@ public Mono<RewardOrganizationExport> retrieveStuckExecution() {
     }
 
     @Override
-    public Mono<RewardOrganizationExport> retrieve() {
+    public Mono<RewardOrganizationExport> retrieve(LocalDate notificationDateToSearch) {
         return rewardOrganizationExportsRepository.reserveExport()
-                .switchIfEmpty(Mono.defer(this::retrieveNewExports))
+                .switchIfEmpty(retrieveNewExports(notificationDateToSearch))
```

Comment:
without the Mono.defer the switchIfEmpty method would be executed immediately when building the reactive pipeline: this time the Mono.defer is required in order to let the "log.info(..)" to be executed only when the empty has been really checked
```suggestion
                .switchIfEmpty(Mono.defer(() => retrieveNewExports(notificationDateToSearch)))
```